opal/mca/accelerator/base/accelerator_base_select.c

```diff
@@ -0,0 +1,148 @@
+} accelerator_opened_component_t;
+
+
+int opal_accelerator_base_select(void)
+{
+
+    int priority = 0, best_priority = -1, ret = 0;
+    opal_list_item_t *item = NULL;
+    mca_base_component_list_item_t *cli = NULL;
+    opal_accelerator_base_component_t *component = NULL, *best_component = NULL;
+    opal_accelerator_base_module_t *module = NULL, *best_module = NULL;
+    opal_list_t opened;
+    accelerator_opened_component_t *aoc = NULL;
+
+    /* Traverse the list of available components; call their init
+       functions. */
+
+    OBJ_CONSTRUCT(&opened, opal_list_t);
+    OPAL_LIST_FOREACH(cli, &opal_accelerator_base_framework.framework_components, mca_base_component_list_item_t) {
+        component = (opal_accelerator_base_component_t *) cli->cli_component;
+        if (NULL == component->accelerator_init) {
+            opal_output_verbose( 10, opal_accelerator_base_framework.framework_output,
+                                 "select: no init function; ignoring component %s",
+                                 component->base_version.mca_component_name );
+            continue;
+        }
+
+        /* Init component to get its priority */
+        opal_output_verbose( 10, opal_accelerator_base_framework.framework_output,
+                             "select: initializing %s component %s",
+                             component->base_version.mca_type_name,
+                             component->base_version.mca_component_name );
+        priority = best_priority;
+        module = component->accelerator_init(&priority);
+        if (NULL == module) {
+            opal_output_verbose( 10, opal_accelerator_base_framework.framework_output,
+                                 "select: init returned failure for component %s",
+                                 component->base_version.mca_component_name );
+            continue;
+        }
+        opal_output_verbose( 10, opal_accelerator_base_framework.framework_output,
+                             "select: init returned priority %d", priority );
+        if (priority > best_priority) {
+            best_priority = priority;
+            best_component = component;
+            best_module = module;
+        }
+        aoc = (accelerator_opened_component_t*)malloc(sizeof(accelerator_opened_component_t));
+        if (NULL == aoc) {
+            return OPAL_ERR_OUT_OF_RESOURCE;
+        }
+        OBJ_CONSTRUCT(aoc, opal_list_item_t);
+        aoc->accelerator_component = component;
+        opal_list_append(&opened, (opal_list_item_t*) aoc);
+    }
+
+    /* Finished querying all components. Check for no components found */
+
+    if( NULL == best_component ) {
+        opal_show_help("help-mca-base.txt", "find-available:none found",
+                       true, "accelerator",
+                       opal_process_info.nodename,
+                       "accelerator");
+    }
+
+    opal_output_verbose( 10, opal_accelerator_base_framework.framework_output,
```

Comment:
be consistent in your files with calling functions as `func()` vs. `func ()`.  Everyone but Nathan appears to prefer `func()`, which is what I would like to see in this PR.

Reply:
Fixed